src/core/Akka/IO/TcpConnection.cs

```diff
@@ -857,6 +858,26 @@ public override PendingWrite DoWrite(ConnectionInfo info)
                 }
             }
 
+            private int Send(ByteString data)
+            {
+                try
+                {
+                    return _connection.Socket.Send(data.Buffers);
+                }
+                catch (SocketException e) when (e.SocketErrorCode == SocketError.WouldBlock)
+                {
+                    try
+                    {
+                        _connection.Socket.Blocking = true;
+                        return _connection.Socket.Send(data.Buffers);
```

Comment:
Looks good to me

Reply:
Guys. Does it mean that connection after that will be "blocking" forever ?

Reply:
No, only for this operation - on the line below we turn the socket back to non-blocking afterwards in the `finally` block. Unless the semantics of how this is implemented by the CLR are different than expected...